server/repository/assignments.go

```diff
@@ -17,17 +17,27 @@ func NewAssignments(db *sql.DB) *Assignments {
 	return &Assignments{db: db}
 }
 
-// ErrNoAssignmentsInitialized is the error returned when the Assignments of a
-// User are requested for a given Experiment, but they have not been yet created
-var ErrNoAssignmentsInitialized = fmt.Errorf("No assignments initialized")
-
 const (
 	insertAssignmentsSQL = `INSERT INTO assignments (user_id, pair_id, experiment_id, answer, duration) VALUES ($1, $2, $3, $4, $5)`
-	selectIDFilePairsSQL = `SELECT id FROM file_pairs WHERE experiment_id=$1`
+	selectIDFilePairsSQL = `SELECT id FROM file_pairs WHERE experiment_id=$1 AND id NOT IN (SELECT pair_id FROM assignments WHERE experiment_id=$1 AND user_id=$2)`
 	selectAssignmentsSQL = `SELECT * FROM assignments WHERE user_id=$1 AND experiment_id=$2`
 	updateAssignmentsSQL = `UPDATE assignments SET answer='%v', duration=%v WHERE id=%v`
+	countPendingIDsSQL   = `SELECT count(id) FROM file_pairs WHERE experiment_id=$1 AND id NOT IN (SELECT pair_id FROM assignments WHERE experiment_id=$1 AND user_id=$2)`
```

Comment:
By looking at these 2 SQL queries \w sub-queries, it makes me thing that if we have any more of those any time soon and need to change them - that would be a good argument for refactoring to use Postgres/Kallax.